api/all/src/main/java/io/opentelemetry/api/trace/SpanBuilder.java

```diff
@@ -233,6 +234,26 @@
    */
   <T> SpanBuilder setAttribute(AttributeKey<T> key, @Nonnull T value);
 
+  /**
+   * Sets attributes to the {@code Span}. If the {@code Span} previously contained a mapping for any
+   * of the keys, the old values are replaced by the specified values.
+   *
+   * @param attributes the attributes
+   * @return this.
+   */
+  default SpanBuilder setAttributes(Attributes attributes) {
+    if (attributes == null || attributes.isEmpty()) {
+      return this;
+    }
+    SpanBuilder span = this;
+    for (Map.Entry<AttributeKey<?>, Object> entry : attributes.asMap().entrySet()) {
+      @SuppressWarnings("unchecked")
+      AttributeKey<Object> attributeKey = (AttributeKey<Object>) entry.getKey();
+      span = span.setAttribute(attributeKey, entry.getValue());
+    }
+    return span;
```

Comment:
Can't you replace this section with this:
```
    attributes.forEach((key, o) -> this.setAttribute((AttributeKey<Object>) key, o));
    return this;
```

With this, we'd need to add an `@SuppressWarnings("unchecked")` to the interface method, which would be a bit odd, I guess. At the very least, we could put this more efficient implementation down in the SDK's implemenation.

Reply:
That annotation doesn't make it into bytecode or javadoc, so maybe it would be ok?

Reply:
Ah, a little better:
```
    attributes.forEach((key, v) -> {
      @SuppressWarnings("unchecked")
      AttributeKey<Object> k = (AttributeKey<Object>) key;
      setAttribute(k, v);
    });
    return this;
```

Reply:
Sure.  I was thinking I'd want to be agnostic about whether `setAttribute` returned the same instance or a different immutable instance but if it's always guaranteed to be the former then it's a moot point.  I could always scope the unchecked operation to an assignment to a local within the lambda which I would think Hotspot should be able to elide.

Reply:
hm. interesting. I suppose there could be an impl that was immutable and this wouldn't work with that. Perhaps just push the better one down into the SDK where we know we're mutable?

Reply:
The javadoc has `@return this`, which I think specifies the contract for this interface that it's always mutable so I think we don't need to worry too much.

Reply:
Good point, I'm being too defensive here.  I'll also update the name of the method to `setAllAttributes` to I think more clearly describe what it does.